src/components/Navigation/Ui/ComponentButtons.js

```diff
@@ -58,88 +58,101 @@ export default class ComponentButtons extends Component {
         {isCurated && <Tag color="green">Curated</Tag>}
         {hasPendingCurations && <Tag color="gold">Pending Curations</Tag>}
-        <ButtonGroup>
-          {!isSourceComponent && !readOnly && !isSourceEmpty && (
-            <ButtonWithTooltip
-              name="addSourceComponent"
-              tip={'Add the definition for source that matches this package'}
-            >
-              <Button className="list-fa-button" onClick={this.addSourceForComponent.bind(this, component)}>
-                <i className="fas fa-code" />
-              </Button>
-            </ButtonWithTooltip>
-          )}
-          {!isDefinitionEmpty && (
-            <ButtonWithTooltip tip={'Dig into this definition'}>
-              <Button
-                className="list-fa-button"
-                onClick={this.inspectComponent.bind(this, currentComponent, definition)}
-              >
-                <i className="fas fa-search" />
-              </Button>
-            </ButtonWithTooltip>
-          )}
-          <CopyUrlButton
-            route={ROUTE_DEFINITIONS}
-            path={component.toPath()}
-            bsStyle="default"
-            className="list-fa-button"
-          />
-          {!hideVersionSelector && (
-            <ButtonWithTooltip tip={'Switch or add other versions of this definition'}>
-              <div>
-                <Dropdown
-                  trigger={['click']}
-                  overlay={
-                    <Menu>
-                      <Menu.Item
-                        data-test-id="switch-component-version"
-                        onClick={this.showVersionSelectorPopup.bind(this, currentComponent, false)}
-                      >
-                        Switch version
-                      </Menu.Item>
-                      <Menu.Item
-                        data-test-id="add-component-version"
-                        onClick={this.showVersionSelectorPopup.bind(this, currentComponent, true)}
-                      >
-                        Add more versions
-                      </Menu.Item>
-                    </Menu>
-                  }
-                >
-                  <Button className="list-fa-button" onClick={event => event.stopPropagation()}>
-                    <i className="fas fa-exchange-alt" /> <Icon type="down" />
-                  </Button>
-                </Dropdown>
-              </div>
-            </ButtonWithTooltip>
-          )}
-          {!readOnly && !isDefinitionEmpty && (
-            <ButtonWithTooltip tip={'Revert Changes of this Definition'}>
-              <Button
-                className="list-fa-button"
-                onClick={() => this.revertComponent(component)}
-                disabled={!hasChange(component)}
-              >
-                <i className="fas fa-undo" />
-              </Button>
-            </ButtonWithTooltip>
-          )}
-        </ButtonGroup>
+        <ButtonToolbar className="hidden-md hidden-lg hidden-sm">
```

Comment:
This would always render this DOM element even if not necessary (when using a desktop view for example).

I suggest to have some kind of control, has done above in the ComponentList for the height of the row, to show a Dropdown or a row of buttons.
I also suggest to move this piece of code in a separate method inside the component.